src/components/timeRangePicker/components/DateHourPicker/components/HourSelector.tsx

```diff
@@ -0,0 +1,108 @@
+import theme from "../../../../../theme";
+import { Button } from "../../../../button";
+import { FlexContainer } from "../../../../container";
+import { Label } from "../../../../input";
+import { SelectButton } from "../../../../selectbutton";
+import { TimeRangeOption } from "../../../types";
+import { UnifiedMultipleSelect, CustomTimeRangeSelector } from "../../styles";
+import { DailyCount } from "../../types";
+import { TimeRangeLabel } from "./TimeRangeLabel";
+
+type Props = {
+  timeRangeOptions: TimeRangeOption[];
+  selectedTimeRange: TimeRangeOption;
+  weekCounts: DailyCount[] | undefined;
+  onSelect: (timeRange: TimeRangeOption, shouldSubmit?: boolean) => void;
+};
+
+export function HourSelector({
+  timeRangeOptions,
+  selectedTimeRange,
+  weekCounts,
+  onSelect,
+}: Props) {
+  const showCustomTimeRange = selectedTimeRange.id === "custom";
+
+  return (
+    <>
+      <Label name="Hours:" htmlFor="time-range-selector" />
+      <FlexContainer
+        gap={theme.spacing.spacing03}
+        id="time-range-selector"
+        alignItems="stretch"
+        style={{ flex: 1 }}
+      >
+        <UnifiedMultipleSelect size="small">
+          {timeRangeOptions.map((timeRangeOption) => (
+            <SelectButton
+              key={timeRangeOption.id}
+              name={timeRangeOption.id}
+              value={timeRangeOption.id}
+              isActive={timeRangeOption.id === selectedTimeRange.id}
+              labelText={
+                <TimeRangeLabel
+                  timeRangeOption={timeRangeOption}
+                  justifyContent={
+                    weekCounts !== undefined ? "space-between" : "center"
+                  }
+                />
+              }
+              onClick={() => onSelect(timeRangeOption, true)}
+            />
+          ))}
+          <SelectButton
+            key="custom"
+            name="custom"
+            value="custom"
+            labelText="custom"
+            isActive={selectedTimeRange.id === "custom"}
+            onClick={() =>
+              onSelect({
+                ...selectedTimeRange,
+                id: "custom",
+              })
+            }
+          />
+        </UnifiedMultipleSelect>
+        <CustomTimeRangeSelector isVisible={showCustomTimeRange}>
+          <Label name="From" htmlFor="time-range-start" />
+          <input
```

Comment:
These inputs should probably be our `Input`, as right now they ignore theming completely
![image](https://user-images.githubusercontent.com/3169290/234845668-cbcc508b-99b7-4cf0-ab5d-5c4061fdd750.png)